captcha/fields.py

```diff
@@ -77,6 +77,15 @@ def validate(self, value):
                 self.error_messages["captcha_error"], code="captcha_error"
             )
 
+        if (
+            isinstance(self.widget, ReCaptchaV3)
+            and check_captcha.action != self.widget.action
+        ):
+            logger.warning("ReCAPTCHA validation failed due to: mismatched action")
+            raise ValidationError(
+                self.error_messages["captcha_invalid"], code="captcha_invalid"
+            )
+
```

Comment:
**Suggestion:** let's move this check after the `is_valid` check below.

The reason for this is that if there is a captcha error there won't be an action set, meaning we will log a warning about a mismatched action and don't log the actual error code returned by ReCaptcha.